gravitee-apim-gateway/gravitee-apim-gateway-services/gravitee-apim-gateway-services-debug/src/main/java/io/gravitee/gateway/debug/policy/impl/PolicyDebugDecorator.java

```diff
@@ -46,12 +50,26 @@ public String id() {
 
     @Override
     public void execute(PolicyChain chain, ExecutionContext context) throws PolicyException {
+        DebugExecutionContext debugContext = (DebugExecutionContext) context;
+
+        DebugStep<?> debugStep = DebugStepFactory.createExecuteDebugStep(policy.id(), streamType, uuid);
+        debugContext.beforePolicy(debugStep);
         policy.execute(chain, context);
+        debugContext.afterPolicy(debugStep);
```

Comment:
In `PolicyChain` you directly call `policy.execute(this, executionContext);`
since `policy` is a `PolicyDebugDecorator` here, we can `beforePolicy` and `afterPolicy` around the delegate policy `execute` method

Reply:
fixed with the introduction of a `DebugPolicyChain` (inspired by `TracingPolicyChain`) in this [commit]([eb345f7](https://github.com/gravitee-io/gravitee-api-management/pull/1219/commits/eb345f7e82eb4231463bfb5748f1feff6dd8b9b4))